python/cmake/thirdparty/get_nvcomp.cmake

```diff
@@ -13,28 +13,19 @@
 # =============================================================================
 
 # This function finds nvcomp and sets any additional necessary environment variables.
-function(find_and_configure_nvcomp VERSION)
+function(find_and_configure_nvcomp)
 
-  # Find or install nvcomp
-  rapids_cpm_find(
-    nvcomp ${VERSION}
-    GLOBAL_TARGETS nvcomp::nvcomp
-    CPM_ARGS GITHUB_REPOSITORY NVIDIA/nvcomp
-    GIT_TAG c435afaf4ba8a8d12f379d688effcb185886cec1
-    OPTIONS "BUILD_STATIC ON" "BUILD_TESTS OFF" "BUILD_BENCHMARKS OFF" "BUILD_EXAMPLES OFF"
+  include(${rapids-cmake-dir}/cpm/nvcomp.cmake)
+  rapids_cpm_nvcomp(
+    BUILD_EXPORT_SET cuspatial-exports
+    INSTALL_EXPORT_SET cuspatial-exports
+    USE_PROPRIETARY_BINARY ${CUDF_USE_PROPRIETARY_NVCOMP}
```

Comment:
This line looks to be a renmant of copy and paste from CUDF. 

Either kvikio should add a cmake option like `KVIKIO_USE_PROPRIETARY_NVCOMP` or you should have a hardcoded value. The reason that cudf provides an option is that nvcomp is closed source and therefore comes with the nvidia closed source license. Therefore users have the choice to not agree to this license and use an older version.

Reply:
This is something very relevant to the work I'm doing in https://github.com/rapidsai/kvikio/pull/120. I need to support the secret build which is currently 2.3.3, and the public build which is currently 2.2.0. I'm not sure now to disambiguate them or support the difference.

Reply:
cudf uses the C++ macro `__has_include` to determine what support the provided nvcomp has.
https://github.com/rapidsai/cudf/blob/branch-22.10/cpp/src/io/comp/nvcomp_adapter.cpp#L24

Reply:
I guess I'll just set `USE_PROPRIETARY_BINARY` to False for now? I don't understand what context I'd want to use the proprietary binary or not, and I don't want to move a licensing agreement onto the kvikio users.

Reply:
If you want to use 2.3+ you will need to enable the proprietary binary.

Reply:
Is `cudf` configured to use the proprietary binary indefinitely?

Reply:
yes